data_structures/binary_trees/words_alphabetical.c

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file
+ * @brief Printing the [words contained in a file](http://www.dailyfreecode.com/Code/word-list-reads-text-file-makes-2050.aspx) named `file.txt` in alphabetical
+ * order and also their frequencies in to another file "wordcount.txt"
+ * @details
+ * More details about this
+ * [algorithm](http://www.dailyfreecode.com/Code/word-list-reads-text-file-makes-2050.aspx)
+ */
+#include <ctype.h>    /// for type checks
+#include <stdbool.h>  /// for boolean data type
+#include <stdio.h>    /// for IO operations
+#include <stdlib.h>   /// for memory allocation
+#include <string.h>   /// for string operations
+
+/**
+ * @brief structure defining a node in the binary tree
+ */
+struct Node
+{
+    char *word;              ///< the word (value) of the node
+    unsigned int frequency;  ///< number of occurences of the word
+    struct Node *left;       ///< pointer to the left child node
+    struct Node *right;      ///< pointer to the right child node
+};
+
+struct Node *readWordsInFileToTree(
+    FILE *file, struct Node *root);  ///< Reads words from file to tree and
+                                     ///< returns pointer to root node
+struct Node *addWordToTree(
+    char *word,
+    struct Node *currentNode);  ///< Adds word (node) to the correct position in
+                                ///< tree and returns pointer to root node
+void writeContentOfTreeToFile(
+    struct Node *node,
+    FILE *file);  ///< Writes contents of tree to another file alphabetically
+char *getPointerToWord(
+    char *word);  ///< Stores a word in memory and returns pointer to word
+struct Node *allocateMemoryForNode();  ///< Reserves memory for new node and
+                                       ///< returns pointer to node
+void closeFile(FILE *file);            ///< Closes file after read or write
+void freeTreeMemory(
+    struct Node *node);  ///< Frees memory when program is terminating
+void endProgramAbruptly(char *errorMessage);  ///< Ends program due to an error
```

Comment:
I would recommend deleting these function prototypes and move the `main` function to the bottom. They also make the code longer, and IMO they're not needed.

Reply:
Ok, will do